File: backend_sprringboot/src/main/java/com/backend_sprringboot/model/UserEntity.java
package com.backend_sprringboot.model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "users")
@Data
public class UserEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "user_id")
    private Integer id;

    @Column(name = "username")
    private String name;

    @Column(name = "email")
    private String email;

    @Column(name = "balance")
    private Double balance;

    @Column(name = "card_number")
    private String cardNumber;

    @Column(name = "bank")
    private String bank;

    @Column(name = "hashed_password")
    private String password;

    @Column(name = "created_at")
    private LocalDateTime created;

    @Column(name = "updated_at")
    private LocalDateTime updated;
    @Column(name = "address")
    private String address;
    @Column(name = "phone")
    private String phone;
    @Column(name = "CCCD_number")
    private String CCCD;
    @OneToMany(mappedBy = "user")
    private List<LoanContractEntity> loanContract;
}

File: lib/models/UserEntity.dart
import 'LoanContractEntity.dart';

class UserEntity {
  int id;
  String name;
  String email;
  double balance;
  String cardNumber;
  String bank;
  String password;
  DateTime created;
  DateTime updated;
  String address;
  String phone;
  String cccd;
  List<LoanContractEntity> loanContract;

  UserEntity({
    required this.id,
    required this.name,
    required this.email,
    required this.balance,
    required this.cardNumber,
    required this.bank,
    required this.password,
    required this.created,
    required this.updated,
    required this.address,
    required this.phone,
    required this.cccd,
    required this.loanContract,
  });

  factory UserEntity.fromJson(Map<String, dynamic> json) {
    return UserEntity(
      id: json['id'],
      name: json['name'],
      email: json['email'],
      balance: json['balance'],
      cardNumber: json['cardNumber'],
      bank: json['bank'],
      password: json['password'],
      created: DateTime.parse(json['created']),
      updated: DateTime.parse(json['updated']),
      address: json['address'],
      phone: json['phone'],
      cccd: json['cccd'],
      loanContract: (json['loanContract'] as List)
          .map((i) => LoanContractEntity.fromJson(i))
          .toList(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'email': email,
      'balance': balance,
      'cardNumber': cardNumber,
      'bank': bank,
      'password': password,
      'created': created.toIso8601String(),
      'updated': updated.toIso8601String(),
      'address': address,
      'phone': phone,
      'cccd': cccd,
      'loanContract': loanContract.map((i) => i.toJson()).toList(),
    };
  }
}